NEW_README.md

```diff
@@ -141,15 +140,155 @@ Run prints its output if either,
+
+##### `id`
+
+The unique identifier for a task. It is used for:
+- Command line invocation (`$ run <id>`)
+- Identifying the task in the interactive UI (TUI).
+
+##### `type`
+
+Specifies the task's lifecycle management strategy:
+- A task can be of only one type: `long` or `short`, any other value is invalid
+- **`long`**: The task is kept alive indefinitely. It's restarted if it exits unexpectedly. **Not suitable as a trigger**.
+  - For example, a development server or a test runner.
+  - If a task depends on a "long" task, Run doesn't really know when the long task has produced whatever output is depended on, so the dependent is run 500ms after the long task starts.
+- **`short`**: The task is considered complete upon successful execution. If it fails, Run will retry it after a short delay.
+  - For example, a build script or a code generation task.
+
+#### Optional Fields
+
+##### `description`
+
+A human-readable description of the task. It's displayed in the TUI task list, the output of `run -list`, and can be used to document the task's purpose. It can be one line or multiline.
+
+##### `dependencies`
+Lists other task IDs that should run alongside this task. 
+
+- If task A depends on task B, B starts before A.
+- If B is a "long" task, A will start 500ms after B starts, and if B is a "short" task, A will start as soon as B completes.[^](#type)
+  
+##### `triggers`
+Lists other task IDs that should run alongside this task, and when completed successfully, cause this task to restart. 
+
+- Triggers are similar to dependencies but specifically for re-execution upon successful completion of the trigger task.
+- If task A is triggered by task B, and both A and B are running, a successful completion of B will trigger A to restart.
+- **Triggers are not considered when a task is initially run.** They only affect task restarts.
+- **Long tasks cannot be triggers.** It is invalid to use a long task as a trigger, since long tasks aren't expected to end.
+
+>[!NOTE]
+> Task IDs listed as `dependencies` and `triggers` can cross directory boundaries (e.g., `"css/build"` refers to a task in `./css/tasks.toml`).
+
+##### `watch`
+Defines file paths or globs to monitor for changes. Any detected change triggers a task restart. Examples include:
+- `"."` for the current directory (excluding subdirectories)
+- `"**"` for any change within the working directory
+- Specific paths like `"./src/website/**/*.js"` for targeted file types.
+
+##### `env`
+A map of environment variables provided to the task's execution environment. 
+
+- These variables are available to the command running the task.
+- They are appended to the current environment, overriding any existing variables with the same name.
+- They persist for the duration of the task's execution.
+
+#### `cmd`
+The shell command to execute as the task. It's run in a new bash process:
```

Comment:
TODO: mention explicitly that cmd is optional, and that cmdless tasks can be useful for grouping